### PR TITLE
Resend community offline updates

### PR DESCRIPTION
**Pull request by vitvly**
Builds on top of https://github.com/status-im/status-go/commit/f1d1726a4e12159013d72008c949fee77e48627c

Added a separate testing suite to deal with sending messages in a community under flaky network conditions.

In order to facilitate testing, an extra method `SetSendFailures` is added to `Waku` struct, and message resend params are configurable via Messenger Options.

Looking into failing tests.

Related to https://github.com/status-im/status-mobile/issues/17950

### CODE REVIEW

File: protocol/persistence.go [hunk 1]
```diff
@@ -856,7 +856,7 @@ func (db sqlitePersistence) ExpiredMessagesIDs(maxSendCount int) ([]string, erro
 			FROM
 				raw_messages
 			WHERE
-			message_type IN (?, ?) AND sent = ? AND send_count <= ?`,
+			(message_type IN (?, ?) OR resend_automatically) AND sent = ? AND send_count <= ?`,
```
Comment (cammellos) [inline protocol/persistence.go:859]: I think we want an index on `resend_automatically` if not there, also I think it's safer for release to do the fix as you did, but for develop we can stop checking the message type, and only trust `resend_automatically`, what do you think?

Comment (vitvly) [reply protocol/persistence.go:859]: Ok, i will add the index for release! As for dropping message_type checks in develop, would it work if i verify them and fix tests (if any broken with this PR) separately?

**Comment by churik** [comment]
@vitvly I tested fix in https://github.com/status-im/status-mobile/pull/17979, and it is working 🎉 
As I see some review comments, can you ping me when PR is approved for a quick re-check?

Thank you!

File: wakuv2/waku.go [hunk 2]
```diff
@@ -995,13 +996,19 @@ func (w *Waku) UnsubscribeMany(ids []string) error {
...
+
 func (w *Waku) broadcast() {
 	for {
 		select {
 		case envelope := <-w.sendQueue:
 			var err error
 			logger := w.logger.With(zap.String("envelopeHash", hexutil.Encode(envelope.Hash())), zap.String("pubsubTopic", envelope.PubsubTopic()), zap.String("contentTopic", envelope.Message().ContentTopic), zap.Int64("timestamp", envelope.Message().Timestamp))
-			if w.settings.LightClient {
+			if w.settings.SendFailures {
+				err = errors.New("Test send failure")
```
Comment (kaichaosun) [inline wakuv2/waku.go:1030]: How about change `SendFailures` to `SkipPublishToTopic` which may feel less test specific, and add a short comment explain it mostly used in test scenarios?
Wondering if `err` is needed here, I haven't look into how `EventEnvelopeExpired` is used.

Comment (vitvly) [reply wakuv2/waku.go:1030]: Good idea, thanks, renamed. The error value at least is useful in tests, it's visible in the output.

File: protocol/persistence.go (hunk 1 above)
Comment (vitvly) [reply protocol/persistence.go:859]: Added index for `resend_automatically`.

**Review by kaichaosun — APPROVED** [review]
LGTM

**Pull request merged**